examples/user_attention/src/main.rs

```diff
@@ -0,0 +1,128 @@
+
+    auto_reset: bool,
+    reset_at: Option<DateTime<Utc>>,
+}
+
+impl Application {
+    fn new(_cc: &CreationContext<'_>) -> Self {
+        Self {
+            attention: UserAttentionType::Informational,
+            request_at: None,
+            auto_reset: false,
+            reset_at: None,
+        }
+    }
+
+    fn attention_reset_timeout() -> Duration {
+        Duration::seconds(3)
+    }
+
+    fn attention_request_timeout() -> Duration {
+        Duration::seconds(2)
+    }
+
+    fn repaint_max_timeout() -> std::time::Duration {
+        std::time::Duration::from_secs(1)
+    }
+}
+
+impl eframe::App for Application {
+    fn update(&mut self, ctx: &Context, frame: &mut eframe::Frame) {
+        if let Some(request_at) = self.request_at {
+            if request_at < Utc::now() {
+                self.request_at = None;
+                frame.request_user_attention(self.attention);
+                if self.auto_reset {
+                    self.auto_reset = false;
+                    self.reset_at = Some(Utc::now() + Self::attention_reset_timeout());
+                }
+            }
+        }
+
+        if let Some(reset_at) = self.reset_at {
+            if reset_at < Utc::now() {
+                self.reset_at = None;
+                frame.request_user_attention(UserAttentionType::Reset);
+            }
+        }
+
+        CentralPanel::default().show(ctx, |ui| {
+            ui.vertical(|ui| {
+                ui.horizontal(|ui| {
+                    ui.label("Attention type:");
+                    eframe::egui::ComboBox::new("attention", "")
+                        .selected_text(repr(self.attention))
+                        .show_ui(ui, |ui| {
+                            for kind in [
+                                UserAttentionType::Informational,
+                                UserAttentionType::Critical,
+                            ] {
+                                ui.selectable_value(&mut self.attention, kind, repr(kind));
+                            }
+                        })
+                });
+
+                let enabled = self.request_at.is_none() && self.reset_at.is_none();
+                let resp = ui
+                    .add_enabled(
+                        enabled,
+                        Button::new(match enabled {
+                            true => format!(
+                                "Request in {} seconds",
+                                Self::attention_request_timeout().num_seconds()
+                            ),
+                            false => match self.reset_at {
+                                None => "Unfocus the window, fast!".to_owned(),
+                                Some(t) => format!(
+                                    "Resetting attention in {} s...",
+                                    (t - Utc::now()).num_seconds()
+                                ),
+                            },
+                        }),
```

Comment:
`match` on bool is more usually (and more tersely) done with `if bool { } else { }`

Reply:
it would also enhance readability if the match was assigned to a temporary `let button_text = if enabled { … } else { … };`